custom_components/sector/event.py

```diff
@@ -44,49 +66,83 @@ class SectorAlarmEvent(SectorAlarmBaseEntity, EventEntity):
+            self._device_name,
+            self._serial_no,
         )
 
     @property
     def event_types(self):
         """Return the list of event types this entity can handle."""
         return ["lock", "unlock", "lock_failed"]
 
-    async def async_update(self):
+    @callback
+    def _async_handle_event(self):
         """Update entity based on the most recent event."""
-        grouped_events = await self.coordinator.process_events()
-        events_for_device = grouped_events.get(self._serial_no)
+        grouped_events = self.coordinator.process_events
+        _LOGGER.debug("SECTOR_EVENT: Processing events for device %s", self._serial_no)
 
-        if not events_for_device:
+        if self._serial_no not in grouped_events:
             _LOGGER.debug(
                 "SECTOR_EVENT: No events found for device %s", self._serial_no
             )
             return
 
-        for event_type, logs in events_for_device.items():
-            latest_log = logs[-1]
-            self._last_event_type = event_type
-            self._events.append(latest_log)
-            self._trigger_event(self._last_event_type, latest_log)
-            self.async_write_ha_state()
+        if self._serial_no in grouped_events:
+            for event_type, logs in grouped_events[self._serial_no].items():
+                latest_log = logs[-1]  # Get the latest log
+                event_time = datetime.fromisoformat(latest_log["time"])
+
+                if (
+                    self._events
+                    and datetime.fromisoformat(self._events[-1]["time"]) >= event_time
+                ):
+                    _LOGGER.debug(
+                        "Skipping update for %s: Event is not newer than the last processed event.",
+                        self._serial_no,
+                    )
+                    continue
+
+                self._last_event_type = event_type
+                self._events.append(latest_log)
+                self._last_formatted_event = latest_log.get(
+                    "formatted_event", f"{event_type.capitalize()} occurred"
+                )
+                event_time = event_time.replace(tzinfo=dt_util.UTC)
+                self._trigger_event(
+                    self._last_event_type, {"timestamp": event_time.isoformat()}
+                )
+                self.async_write_ha_state()
+
+                _LOGGER.debug(
+                    "SECTOR_EVENT: Entity %s updated with new event: type=%s, time=%s",
+                    self._attr_unique_id,
+                    self._last_event_type,
+                    event_time,
+                )
 
     def _trigger_event(self, event_type, event_attributes):
-        """Trigger an event with timestamp and details."""
+        """Trigger an event update with the latest type."""
         event_timestamp = event_attributes.get(
-            "Time", datetime.now(timezone.utc).isoformat()
+            "timestamp", datetime.now(dt_util.UTC).isoformat()
```

Comment:
I think we should just set the correct time in `__last_event_triggered` as mentioned also below, then remove the formatting etc. and let it be done by the built in.